src/redux/actions/genes/loadMarkerGenes.js

```diff
@@ -0,0 +1,55 @@
+import {
+  MARKER_GENES_ERROR, MARKER_GENES_LOADING, MARKER_GENES_LOADED,
+} from '../../actionTypes/genes';
+
+import { fetchCachedWork } from '../../../utils/cacheRequest';
+
+const REQUEST_TIMEOUT = 60;
+
+const loadMarkerGenes = (experimentId, resolution) => async (dispatch, getState) => {
+  // Disabled linter because we are using == to check for both null and undefined values
+  // eslint-disable-next-line eqeqeq
+  if (experimentId == null || resolution == null) throw new Error('Null or undefined parameter/s for loadMarkerGenes');
+
+  const { backendStatus, processing } = getState().experimentSettings;
+
+  const { method } = processing.configureEmbedding.clusteringSettings;
+
+  const body = {
+    name: 'MarkerHeatmap',
+    nGenes: 2,
+    type: method,
+    config: {
+      resolution,
+    },
+  };
+
+  dispatch({
+    type: MARKER_GENES_LOADING,
+  });
+
+  try {
+    const data = await fetchCachedWork(experimentId, REQUEST_TIMEOUT, body, backendStatus.status);
```

Comment:
A small thing that you can choose to ignore: if we tend to set the same value for `REQUEST_TIMEOUT` for most places where we call `fetchCachedWork`, we may want to have a default value for timeout set in the actual `fetchCachedWork` function definition instead of specifying it everywhere.